gc/base/MemoryPoolAddressOrderedList.hpp

```diff
@@ -75,9 +82,26 @@ class MM_MemoryPoolAddressOrderedList : public MM_MemoryPoolAddressOrderedListBa
 	void updateHintsBeyondEntry(MM_HeapLinkedFreeHeader *freeEntry);
 	void *internalAllocate(MM_EnvironmentBase *env, uintptr_t sizeInBytesRequired, bool lockingRequired, MM_LargeObjectAllocateStats *largeObjectAllocateStats);
 	bool internalAllocateTLH(MM_EnvironmentBase *env, uintptr_t maximumSizeInBytesRequired, void * &addrBase, void * &addrTop, bool lockingRequired, MM_LargeObjectAllocateStats *largeObjectAllocateStats);
+	MMINLINE bool doesNeedAlignment(MM_EnvironmentBase *env, MM_HeapLinkedFreeHeader *freeEntry)
+	{
+		bool ret = false;
+		if ((NULL != _firstUnAlignmentFreeEntry) && (NULL != freeEntry) && ((uintptr_t) freeEntry >= (uintptr_t)_firstUnAlignmentFreeEntry)) {
```

Comment:
I hope we should be able to remove the NULL checks, if we keep _firstUnAlignmentFreeEntry at max value (or at the end of heap/area/pool, if the end is known) in case the pool is already fully aligned.